src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.dsde.workbench.sam.service
+
+import akka.http.scaladsl.model.StatusCodes
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam._
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/**
+  * Created by gpolumbo on 2/21/2018.
+  */
+class ManagedGroupService(resourceService: ResourceService, val resourceTypes: Map[ResourceTypeName, ResourceType]) extends LazyLogging {
+
+  def managedGroupType: ResourceType = resourceTypes.getOrElse(ManagedGroupService.ManagedGroupTypeName, throw new WorkbenchException(s"resource type ${ManagedGroupService.ManagedGroupTypeName.value} not found"))
+  def memberRole = managedGroupType.roles.find(_.roleName == ManagedGroupService.MemberRoleName).getOrElse(throw new WorkbenchException(s"${ManagedGroupService.MemberRoleName} role does not exist in $managedGroupType"))
+
+  // Important Note, see: org/broadinstitute/dsde/workbench/sam/model/SamModel.scala:90
+  def createManagedGroup(groupId: ResourceId, userInfo: UserInfo): Future[Resource] = {
+    for {
+      managedGroup <- resourceService.createResource(managedGroupType, groupId, userInfo)
+      _ <- createPolicyForMembers(managedGroup)
+      policies <- resourceService.accessPolicyDAO.listAccessPolicies(managedGroup)
+      _ <- createAggregateGroup(managedGroup, policies)
+    } yield managedGroup
+  }
+
+  private def createPolicyForMembers(managedGroup: Resource): Future[AccessPolicy] = {
+    val accessPolicyName = AccessPolicyName(memberRole.roleName.value)
+    val resourceAndPolicyName = ResourceAndPolicyName(managedGroup, accessPolicyName)
+    resourceService.createPolicy(resourceAndPolicyName, members = Set.empty, Set(memberRole.roleName), actions = Set.empty)
+  }
+
+  private def createAggregateGroup(resource: Resource, componentPolicies: Set[AccessPolicy]): Future[BasicWorkbenchGroup] = {
+    val email = generateManagedGroupEmail(resource.resourceId)
+    val workbenchGroupName = WorkbenchGroupName(resource.resourceId.value)
+    val groupMembers: Set[WorkbenchSubject] = componentPolicies.map(_.id)
+    resourceService.directoryDAO.createGroup(BasicWorkbenchGroup(workbenchGroupName, groupMembers, email))
+  }
+
+  private def generateManagedGroupEmail(resourceId: ResourceId): WorkbenchEmail = {
+    val localPart = resourceId.value
```

Comment:
I prefer the term groupName instead of localPart

Reply:
fixed